src/panel-page.tsx

```diff
@@ -52,15 +68,64 @@ function PagerLadderHelp(props: { room: PageRoom }) {
 	);
 }
 
+function PageServerHTML(props: { room: PageRoom }) {
+	const { room } = props;
+	if (room.loading) {
+		return <p>Loading...</p>;
+	} else {
+		return <div class="page-html-container">
+			<SanitizedHTML>{room.htmlData || ''}</SanitizedHTML>
+		</div>;
+	}
+}
+
 class PagePanel extends PSRoomPanel<PageRoom> {
+	clientRooms: { [key: string]: JSX.Element } = { 'ladderhelp': <PageLadderHelp room={this.props.room}/> };
+
+	/**
+	 * @return true to prevent line from being sent to server
+	 */
+	receiveLine(args: Args) {
+		const { room } = this.props;
+		switch (args[0]) {
+		case 'title':
+			room.title = args[1];
+			PS.update();
+			return true;
+		case 'tempnotify': {
+			const [, id, title, body] = args;
+			room.notify({title, body, id});
+			return true;
+		}
+		case 'tempnotifyoff': {
+			const [, id] = args;
+			room.dismissNotification(id);
+			return true;
+		}
+		case 'selectorhtml':
+			const pageHTMLContainer = this.base!.querySelector('.page-html-container');
+			const selectedElement = pageHTMLContainer?.querySelector(args[1]);
+			if (!selectedElement) return;
+			selectedElement.innerHTML = BattleLog.sanitizeHTML(args.slice(2).join('|'));
+			room.isSubtleNotifying = true;
+			return true;
+		case 'noinit':
+			if (args[1] === 'namerequired') {
+				room.setHTMLData(args[2]);
+			}
+			return true;
+		case 'pagehtml':
+			room.setHTMLData(args[1]);
+			return true;
+		}
+	}
 	render() {
 		const { room } = this.props;
```

Comment:
```suggestion
		const {room} = this.props;
```

Reply:
We don't have tslint but you know the rules and so do I!

Reply:
I have made these changes, but what are the rules exactly?